File: Stock/src/main/java/net/lahlalia/stock/entities/Depot.java
package net.lahlalia.stock.entities;

import jakarta.persistence.*;
import lombok.*;
import net.lahlalia.stock.enums.Area;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
public class Depot {

    @Id
    private String idDepot;
    private String nameDepot;
    private String zone;
    @Enumerated(EnumType.STRING)
    private Area area;

    @OneToMany(mappedBy = "depot", cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<Bac> bacs;

    @OneToMany(mappedBy = "depot", cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<HistoryStock> historyStocks;





}

File: vivo-frontend/src/app/models/Depot.ts
import {Bac} from "./Bac";
import {BacDto} from "./BacDto";

export class Depot{
  idDepot !: string;
  nameDepot: String|undefined;
  zone: String|undefined;
  area : any|undefined;
  bacDtos: BacDto[]|undefined;
  stock: number|undefined;


}
